src/preamble.js

```diff
@@ -2249,6 +2249,15 @@ function integrateWasmJS() {
       }
     }
 
+    function isD8Shell() {
+      // d8 shell doesn't always resolve the WebAssembly.instantiate promise
+      // when exiting, so we need to call testRunner.waitUntilDone/.notifyDone,
+      // which only exist in d8
+      return (ENVIRONMENT_IS_SHELL && testRunner &&
```

Comment:
what is this `testRunner`?

If it's a d8-only thing, it will throw in other shells. Can check if it exists first (`typeof testRunner === 'object'` or such).

Reply:
I do guard with `&& testRunner`, but that's kind of hard to see buried in there?
For my knowledge, is `typeof testRunner === 'object'` safer, or just more precise about its expectations?

Reply:
it behaves differently: `testRunner` will throw a JS exception if that variable is not declared anywhere. `typeof testRunner` will not throw, and will return `undefined` if it is not declared (it will also return that if it is declared and has that value...